### DISCUSSION TITLE
Troubleshooting ARM after installation

### DISCUSSION THREAD

**Discussion by HaroldRobi** [automatic-ripping-machine/automatic-ripping-machine#675] [Q&A]
I was able to get ARM successfully installed to Ubuntu server 20.04.  Everything works except it will not RIP a dvd,  If I insert a DVD ARM will read the diks, open a log with the name of the DVD but no log information is in it.

Please advise

Thank you

**Comment by microtechno9000** [comment]
In the log folder there is a WebUi.log, can you post that log here.
Also, in the ARM settings please make sure you have it set to Debug

**Reply by HaroldRobi** [reply]
No problem.  Here is  the WebUI.log file you requested
[WebUI.log](https://github.com/automatic-ripping-machine/automatic-ripping-machine/files/10471673/WebUI.log)


The settings file in the webUI look like default settings.  I am going to check the settings from the manual setup to see if my changes stuck or I forgot to save the file.

**Reply by HaroldRobi** [reply]
So I just checked the arm.yaml file and its set to default values.  I will go thorugh and set these options through the web interface.  Seems the manual instructions do not cover setting this file up.

**Reply by HaroldRobi** [reply]
updates to the settings file have been made.  I have verified that the settings I made in the ARM WEBUI were identical to the /opt/arm/arm.yaml.  DEUG has been enabled.

**Reply by microtechno9000** [reply]
Thanks for loading the WebUI log and checking the config. it looks like a newer version of SQLAlchemy has been installed. If you roll it back a version that should fix the install.

not sure why it is pulling in a newer version. will check when I get back to my PC.

**Reply by HaroldRobi** [reply]
According th the install instructions SQLAlchemy is not one of the packages that was installed.  I did a text search of settign up Arm manually and SQLAlchemy is not one of the packages I installed.

It seems after some research the package python3-sqlalchemy is missing, i will be instlaling this package to see if this resolves my issue.

This package is not asked for in the manual setup of ARM.

... after some time

the package python3-sqlalchemy_1.3.12+ds1-1ubuntu2_all.deb was retreived and installed.

I will reboot the server to see if this resolves the issue.

After all that, This did not resolve my issue.

Regards,

Harold Robinson

**Reply by microtechno9000** [reply]
You are correct, there have been some issues of late with that and some other packages with updates. I can't see anything obvious, @shitwolfymakes  or @1337-server , anythoughts

**Reply by HaroldRobi** [reply]
I am following up on this issue.  Thanks again for all your help on this project.

**Reply by microtechno9000** [reply]
Are you still having the issue?

**Reply by HaroldRobi** [reply]
Yes I am no none except you has gotten back to me yet.

**Reply by microtechno9000** [reply]
Sorry about that, there are not many devs looking after ARM.
Have you gone through the [wiki FAQ](https://github.com/automatic-ripping-machine/automatic-ripping-machine/wiki/FAQ)? 

Reading through your comment again, it could be a permission issue. I presume you installed using the automatic script? if not, try running that, it will pull the latest ARM and ensure that all the permissions are setup correctly. [Automatic Installation](https://github.com/automatic-ripping-machine/automatic-ripping-machine/wiki/Ubuntu-20.04-Automatic-Installation)

**Comment by 1337-server** [comment]
Can you give more details ?


What system are you running ?

How did you install ? (Script or manually)

What are the permissions of both your log folder and media folders ? (`ls -la /log/files/here` should give you full details)

Does this happen only with DVD's or does it happen with every disc type ?

Have you checked the syslog to see if ARM is outputting any error messages before it can run ? 
Arm usually spits out some errors messages to the syslog if something isn't right.

**Reply by HaroldRobi** [reply]
I apologize for the late response.

**Can you give more details ?** I insert a DVD disk (its the only drive I have for this new server at this time)  It reads the disk and creates a log.  The log is blank and it never decodes the DVD. 

**How did you install ?** (Script or manually) I instllaed manually following the steps very carefully. Located here:  https://github.com/automatic-ripping-machine/automatic-ripping-machine/wiki/Setting-up-ARM-manually-(Ubuntu)

**What are the permissions of both your log folder and media folders ? (ls -la /log/files/here should give you full details)** - drwxrwxr-x 5 arm  arm  4.0K Jan 15 15:23 media
drwxrwxr-x 2 arm  arm  4.0K Jan 22 00:12 logs

I did receive this error in mail:

Message 7:
From arm@ripasaurus-2.hundredaw.com  Mon Feb 20 12:12:09 2023
X-Original-To: arm
Subject: Output from your job       35
To: arm@ripasaurus-2.hundredaw.com
Date: Mon, 20 Feb 2023 20:12:08 +0000 (UTC)
From: arm@ripasaurus-2.hundredaw.com

Traceback (most recent call last):
  File "/opt/arm/arm/ripper/main.py", line 443, in <module>
    running_jobs = db.session.query(Job).filter(Job.status.notin_(['fail', 'success']), Job.devpath == devpath).all()
  File "<string>", line 2, in query
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/orm/scoping.py", line 24, in _proxied
    return self.registry()
  File "/usr/local/lib/python3.8/dist-packages/sqlalchemy/util/_collections.py", line 1006, in __call__
    key = self.scopefunc()
  File "/usr/local/lib/python3.8/dist-packages/flask_sqlalchemy/session.py", line 81, in _app_ctx_id
    return id(app_ctx._get_current_object())  # type: ignore[attr-defined]
  File "/usr/local/lib/python3.8/dist-packages/werkzeug/local.py", line 513, in _get_current_object
    raise RuntimeError(unbound_message) from None
RuntimeError: Working outside of application context.

This typically means that you attempted to use functionality that needed
the current application. To solve this, set up an application context
with app.app_context(). See the documentation for more information.

**Reply by 1337-server** [reply]
Are you running the latest version of ARM ?

This issue has been tracked before here:
https://github.com/automatic-ripping-machine/automatic-ripping-machine/issues/639

If you are on the latest version, please ping there and ill reopen and look into this

A user did provide a fix here, but this is stale as they submitted a PR to the wrong branch
https://github.com/automatic-ripping-machine/automatic-ripping-machine/pull/640

**Reply by HaroldRobi** [reply]
I don't have a lot of time invested in this install.  I noticed that the manual install is no longer supported.  Shall I do a Docker install instaed?

**Reply by 1337-server** [reply]
Yes, it's much simpler to manage.

**Reply by HaroldRobi** [reply]
Ok, I will do that then and report back.  Thank you very much

**Reply by HaroldRobi** [reply]
The instalaltion instructions are not very clear.  

1.  Install Ubuntu Linux
2. Update Ubuntu (not in the instructions)
3. create user arm
4. then pull down the install script.  and run that..  There are no other packages I need to install before performing that step?

**Reply by 1337-server** [reply]
Nope, that's all. Unless you want HW encoding. But that doesn't work with docker currently. So that's all the steps you need to take.

**Reply by HaroldRobi** [reply]
Am i going to install ubuntu server or desktop?

**Comment by HaroldRobi** [comment]
Ok, everything went very well.  I have ARM installed in a docker successfully.  It is currently ripping its first DVD.  How do I add MakeMKV license to the docker image.  It is needed to rip Blurays.

**Reply by 1337-server** [reply]
You need to add it as a docker env variable when you create the container

`    -e MAKEMKV_APP_KEY="YOU_CODE_GOES_HERE" \`

Add it to your version of this
https://github.com/automatic-ripping-machine/automatic-ripping-machine/blob/main/scripts/docker/start_arm_container.sh

**Reply by HaroldRobi** [reply]
Thank you.

**Comment by HaroldRobi** [comment]
I was looking at my start_arm_container.sh because I was not finding the completed file after the ripping was complete.  I think I might have misconfigured this file.  In the file it is stated `-v "<path_to_arm_user_home_folder>:/home/arm" \` as how the folders are configured.  I just wanted the defaults as that was ok at that point, so I just deleted the <path_to_arm_user_home_folder> so the line looked like this:  `-v "/home/arm" \` but I am guessing that is wrong.  Saying I wanted to keep the default configuration, should it look like this?:  `-v "/home/arm:/home/arm" \` instead?  If the directories have not been created, will the script create the directories for me? If not then I need to create the directories.  If you have to create the folders, permissions will be 775 with owner as **arm**?

BTW I love all the effort you have put into this project.  The new docker method, is much easier to work with your project now.  Thanks very much.

**Comment by 1337-server** [comment]
This would be the correct/default directory
`-v "/home/arm:/home/arm" \`
The way I remember it is with containers is
`Outside` `door` `inside`

ARM should now create all the folders it needs.

You're welcome.

**Comment by HaroldRobi** [comment]
So after making the correct changes.  I have ARM up and running with Docker.  If I wanted to add more DVD/Bluray drives what must I do?

**Reply by 1337-server** [reply]
If you wanted this to last for a long time then you would need to manually build the image from source after making these changes: https://github.com/automatic-ripping-machine/automatic-ripping-machine/commit/4e7227a0a946f42d6251fa080a13785097a2cf42

But if you were happy to re-enter commands now and again (every time you recreate the contiainer) the only things that need added if you have the --priviledged flag is

The mountpount inside the container
`/mnt/dev/sr17`
`/mnt/dev/sr18`
`/mnt/dev/sr19`
`/mnt/dev/sr20` etc

and this line for each drive added to the /etc/fstab with the sr9 being changed for the correct drive/mount point again inside the container.
`/dev/sr9  /mnt/dev/sr9  udf,iso9660  users,noauto,exec,utf8,ro  0  0`

**both INSIDE THE CONTAINER. nothing needs touched outside on the host machine.**

**Reply by HaroldRobi** [reply]
How many drives are supported out of the "box"
